javascript/javascript2/week2/hyfBay/main.js

```diff
@@ -0,0 +1,71 @@
+const products = getAvailableProducts();
+console.log(products);
+// render the prodects to the html page
+const renderProducts = (array) => {
+  let mainTag = document.querySelector("#maintag");
+  mainTag.innerHTML = array
+    .map((item) => {
+      // deconstruction of the array
+      let { name, price, rating } = item;
+      return `<div class='product'>
+    <h2>${name}</h2>
+    <p>price: ${price}</p>
+    <p>rating: ${rating}</p></div><hr/>`;
+    })
+    .join("");
+};
+
+renderProducts(products);
+
+const search = document.querySelector("#search");
+const maxPrice = document.querySelector("#maxprice");
+const button = document.querySelector("#button");
```

Comment:
If you are selecting elements by their ID, getElementById is likely to be faster and more efficient. If you need to select elements based on more complex criteria, querySelector may be the better choice.